tests/compilation_cache_test.py

```diff
@@ -0,0 +1,120 @@
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+import hashlib
+from jax.experimental.compilation_cache import compilation_cache as cc
+import jax
+import jax.test_util as jtu
+import numpy as np
+
+class CompilationCacheTest(jtu.JaxTestCase):
+
+  def test_cache_key_type(self):
+    computation = jax.xla_computation(lambda x, y: x + y)(1, 1)
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    self.assertEqual(type(cc.get_cache_key(computation, compile_options)), str)
+
+  def test_compile_options(self):
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    compile_options_filled = self.fill_compile_options()
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_compile_options, compile_options_filled)
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_compile_options, compile_options_filled)
+    hash_obj3 = self.create_sudo_hash_obj(cc._hash_compile_options, compile_options_not_filled)
+    self.assertEqual(hash_obj1.hex(), hash_obj2.hex())
+    self.assertNotEqual(hash_obj1.hex(), hash_obj3.hex())
+
+  def test_executable_build_options(self):
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    compile_options_filled = self.fill_compile_options()
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_executable_build_options, compile_options_filled.executable_build_options)
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_executable_build_options, compile_options_filled.executable_build_options)
+    hash_obj3 = self.create_sudo_hash_obj(cc._hash_executable_build_options, compile_options_not_filled.executable_build_options)
+    self.assertEqual(hash_obj1.hex(), hash_obj2.hex())
+    self.assertNotEqual(hash_obj1.hex(), hash_obj3.hex())
+
+  def test_debug_options(self):
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_debug_options, compile_options.executable_build_options.debug_options)
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_debug_options, compile_options.executable_build_options.debug_options)
+    self.assertEqual(hash_obj1.hex(), hash_obj2.hex())
+
+  def test_libtpu(self):
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_libtpu, jax.lib.xla_bridge.get_backend())
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_libtpu, jax.lib.xla_bridge.get_backend())
+    self.assertEqual(hash_obj1.hex(), hash_obj2.hex())
```

Comment:
If this test is running on a non-CPU backend, you can get the CPU backend to check the non-equal case! Something like this:
```python
if jax.lib.xla_bridge.get_backend().platform != "cpu":
  cpu_backend = jax.lib.xla_bridge.get_backend("cpu")
  <create hash of cpu backend>
  <assertNotEqual to non-cpu backend hash>
```
You need the if statement to avoid accidentally comparing the same backend and asserting it's not equal when it is.

Reply:
I'm seeing now that the project doc only mentions hashing the platform version value as part of the key...would you like me to take out the other things I am hashing for the libtpu in my main file?

Reply:
No I think it's good! I only had the platform_version in mind initially, but the other things don't hurt, and will be helpful if we ever extend the cache to work on non-TPU backends.